cloud-consumption/src/main/java/com/sequenceiq/consumption/job/storage/StorageConsumptionJobService.java

```diff
@@ -0,0 +1,110 @@
+        if (storageConsumptionConfig.isStorageConsumptionEnabled()) {
+            JobDetail jobDetail = buildJobDetail(resource);
+            Trigger trigger = buildJobTrigger(jobDetail);
+            try {
+                JobKey jobKey = JobKey.jobKey(resource.getJobResource().getLocalId(), JOB_GROUP);
+                if (scheduler.getJobDetail(jobKey) != null) {
+                    LOGGER.info("Unscheduling storage consumption job key: '{}' and group: '{}'", jobKey.getName(), jobKey.getGroup());
+                    unschedule(jobKey.getName());
+                }
+                LOGGER.info("Scheduling storage consumption job for key: '{}' and group: '{}'", jobKey.getName(), jobKey.getGroup());
+                scheduler.scheduleJob(jobDetail, trigger);
+            } catch (SchedulerException e) {
+                LOGGER.error(String.format("Error during scheduling quartz job: %s", jobDetail), e);
+            }
+        }
+
+    }
+
+    public void schedule(Long id) {
+        StorageConsumptionJobAdapter resourceAdapter = new StorageConsumptionJobAdapter(id, applicationContext);
+        schedule(resourceAdapter);
+    }
+
+    private JobDetail buildJobDetail(StorageConsumptionJobAdapter resource) {
+        JobDataMap jobDataMap = resource.toJobDataMap();
+        return JobBuilder.newJob(StorageConsumptionJob.class)
+                .withIdentity(resource.getJobResource().getLocalId(), JOB_GROUP)
+                .withDescription("Getting storage usage")
+                .usingJobData(jobDataMap)
+                .storeDurably()
+                .build();
+
+    }
+
+    private Trigger buildJobTrigger(JobDetail jobDetail) {
+        return TriggerBuilder.newTrigger()
+                .forJob(jobDetail)
+                .withIdentity(jobDetail.getKey().getName(), TRIGGER_GROUP)
+                .withDescription("Trigger for getting storage usage")
+                .startAt(delayedFirstStart())
+                .withSchedule(SimpleScheduleBuilder.simpleSchedule()
+                        .withIntervalInMinutes(storageConsumptionConfig.getIntervalInMinutes())
+                        .repeatForever()
+                        .withMisfireHandlingInstructionIgnoreMisfires())
+                .build();
+
+    }
+
+    public void unschedule(String id) {
+        JobKey jobKey = JobKey.jobKey(id, JOB_GROUP);
+        try {
+            LOGGER.info("Unscheduling storage consumption job key: '{}' and group: '{}'", jobKey.getName(), jobKey.getGroup());
+            scheduler.deleteJob(jobKey);
+        } catch (SchedulerException e) {
+            LOGGER.error(String.format("Error during unscheduling quartz job: %s", jobKey), e);
+        }
+
+    }
+
+    private Date delayedFirstStart() {
+        return Date.from(ZonedDateTime.now().toInstant().plus(Duration.ofSeconds(
+                TimeUnit.MINUTES.toSeconds(RANDOM.nextInt(storageConsumptionConfig.getIntervalInMinutes())))));
```

Comment:
Semantically it's still the same, because this way the random integer is generated from 0-30 and it's just converted to seconds.
```
int delayInSeconds = RANDOM.nextInt((int) TimeUnit.MINUTES.toSeconds(storageConsumptionConfig.getIntervalInMinutes()));
return Date.from(ZonedDateTime.now().toInstant().plus(Duration.ofSeconds(delayInSeconds)));
```
By swapping the conversion and random generation the generated int will be in the 0-1800 range, spreading the starts more evenly with higher granularity.

Reply:
I pushed again with change